### PR TITLE
Singleton for web-common settings

### PR DESCRIPTION
**Pull request by dillon-courts**
The goal of this PR is to allow a web-common user to pass in the required environment variables at initialization instead of having the required env variables set.

This was a much larger undertaking than you would think due to the way environment variable usage was scattered throughout the library.

I used a 'singleton pattern' to create a global object that holds the values of the settings that are used throughout the library. This involved quite a few changes as well as unit test updates.

### CODE REVIEW

File: Crossroads.Web.Common.Test/Configuration/ConfigurationWrapperTest.cs [hunk 1]
```diff
@@ -65,16 +49,18 @@ public void SetUp()
...
+            _authenticationRepository.Setup(mocked => mocked.AuthenticateClient(It.IsAny<string>(), It.IsAny<string>())).Returns(new AuthToken
+            {
+                AccessToken = "token"
+            });
 
             _fixture = new ConfigurationWrapper(_ministryPlatformRest.Object, _authenticationRepository.Object);
         }
 
         [TearDown]
         public void TearDown()
```
Comment (jpc11623) [inline Crossroads.Web.Common.Test/Configuration/ConfigurationWrapperTest.cs]: Does the empty function do anything now?

File: Crossroads.Web.Common.Test/Configuration/WebCommonSettingsTest.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+            string authServiceBaseUrl = "testauthservicebaseurl";
+            string mpOauthBaseUrl = "testmpoauthbaseurl";
+            string mpRestApiEndpoint = "testmprestapiendpoint";
+            string crdsMpCommonClientId = "testcommonclientid";
+            string crdsMpCommonClientSecret = "testcommonclientsecret";
+            string crdsMpApiUser = "testapiuser";
+            string crdsMpApiUserPassword = "testapiuserpassowrd";
+
+            Environment.SetEnvironmentVariable("AUTH_SERVICE_BASE_URL", authServiceBaseUrl);
```
Comment (jpc11623) [inline Crossroads.Web.Common.Test/Configuration/WebCommonSettingsTest.cs]: Would these value persist between test runs?

File: Crossroads.Web.Common.Test/MinistryPlatform/ApiUserRepositoryTest.cs [hunk 3]
```diff
@@ -22,23 +22,25 @@ public class ApiUserRepositoryTest
...
         private MockHttpMessageHandler _mpMessageHandler;
 
-        private const string _commonClientId = "common client Id";
-        private const string _commonClientSecret = "common client secret";
-
         [SetUp]
         public void SetUp()
         {
+            WebCommonSettings.SetSettings(new WebCommonSettingsConfig(
+                "who cares",
```
Comment (jpc11623) [inline Crossroads.Web.Common.Test/MinistryPlatform/ApiUserRepositoryTest.cs]: I care.

File: Crossroads.Web.Common.Test/Configuration/WebCommonSettingsTest.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            Environment.SetEnvironmentVariable("MP_OAUTH_BASE_URL", mpOauthBaseUrl);
+            Environment.SetEnvironmentVariable("MP_REST_API_ENDPOINT", mpRestApiEndpoint);
+            Environment.SetEnvironmentVariable("CRDS_MP_COMMON_CLIENT_ID", crdsMpCommonClientId);
+            Environment.SetEnvironmentVariable("CRDS_MP_COMMON_CLIENT_SECRET", crdsMpCommonClientSecret);
+            Environment.SetEnvironmentVariable("API_USER", crdsMpApiUser);
+            Environment.SetEnvironmentVariable("API_PASSWORD", crdsMpApiUserPassword);
+            
+            Assert.AreEqual(authServiceBaseUrl, WebCommonSettings.Instance.AuthServiceBaseUrl);
```
Comment (jpc11623) [inline Crossroads.Web.Common.Test/Configuration/WebCommonSettingsTest.cs]: These asserts should probably be checking against a literal value, not a variable.

File: Crossroads.Web.Common.Test/Configuration/ConfigurationWrapperTest.cs (hunk 1 above)
Comment (dillon-courts) [reply Crossroads.Web.Common.Test/Configuration/ConfigurationWrapperTest.cs]: I'm not sure I follow this question?

File: Crossroads.Web.Common.Test/Configuration/WebCommonSettingsTest.cs (hunk 2 above)
Comment (dillon-courts) [reply Crossroads.Web.Common.Test/Configuration/WebCommonSettingsTest.cs]: I don't believe so, based on the tests I've run.

File: Crossroads.Web.Common.Test/Configuration/WebCommonSettingsTest.cs (hunk 4 above)
Comment (dillon-courts) [reply Crossroads.Web.Common.Test/Configuration/WebCommonSettingsTest.cs]: Explain why, I feel like that just means I have to re-type a bunch of string literals.

File: Crossroads.Web.Common.Test/Configuration/ConfigurationWrapperTest.cs (hunk 1 above)
Comment (jpc11623) [reply Crossroads.Web.Common.Test/Configuration/ConfigurationWrapperTest.cs]: The TearDown() function appears empty when viewing the full file - https://github.com/crdschurch/crds-web-common/blob/be89eeccb3946b4add77171ceca0f2ae8d41198e/Crossroads.Web.Common.Test/Configuration/ConfigurationWrapperTest.cs

File: Crossroads.Web.Common.Test/Configuration/WebCommonSettingsTest.cs (hunk 4 above)
Comment (jpc11623) [reply Crossroads.Web.Common.Test/Configuration/WebCommonSettingsTest.cs]: It's good style and makes tests more robust. You're trying to compare a test result against something which is as immutable as possible in the code. There's no guarantee someone won't come along and change the value. Additionally, you're essentially comparing it against itself the way it is being used in the test.

**Pull request merged**